src/ast.cc

```diff
@@ -435,6 +414,27 @@ static void destroy_module_field(ModuleField* field) {
   }
 }
 
+Module::Module()
+    : first_field(nullptr),
+      last_field(nullptr),
+      num_func_imports(0),
+      num_table_imports(0),
+      num_memory_imports(0),
+      num_global_imports(0),
+      start(0) {
+  WABT_ZERO_MEMORY(loc);
+  WABT_ZERO_MEMORY(name);
+  WABT_ZERO_MEMORY(funcs);
+  WABT_ZERO_MEMORY(globals);
+  WABT_ZERO_MEMORY(imports);
+  WABT_ZERO_MEMORY(exports);
+  WABT_ZERO_MEMORY(func_types);
+  WABT_ZERO_MEMORY(tables);
+  WABT_ZERO_MEMORY(elem_segments);
+  WABT_ZERO_MEMORY(memories);
+  WABT_ZERO_MEMORY(data_segments);
+}
+
 void destroy_module(Module* module) {
```

Comment:
Not knowing your plans, shouldn't this be converted to a destructor for class Module? This would clean up destruction. It would also simplify destruction of pointers to just a call to "delete".

Note: I understand you are in the process of (slowly) changing things. Therefore this may already be on your road map. Just bringing it up because the code now is somewhat "ugly" and confusing. That is, it isn't immediately obvious when to use:

      destroy_FOO(&F...)

and when to use:

    destroy_FOO(F...); delete F;

The current set up is forcing you to "copy" (non-obvious) destructor sequences to everything going out of scope, rather then letting C++ destructor rules automatically handle it.